### DISCUSSION TITLE
Logging data with the CarCaptured Timestamp

### DISCUSSION THREAD

**Discussion by RaimondB** [tillsteinbach/WeConnect-mqtt#101] [Q&A]
I want to use your docker container to log data about my car to influxDB.
I am getting data on the MQTT broker, but now I am wondering how to achieve that I am using the correct timestamp.

WeConnect-mqtt will pull and publish new data every 300 seconds, but it could be that carCapturedTimestamp still indicates old data. I want to use the carCapturedTimestamp as the timestamp for logging to influx.

But then I need to make sure the timestamp is updated first on the MQTT before the actual measurements. Otherwise, I will we writing the new data using the old timestamp.

So my question is if I can be sure the carCapturedTimestamp will be updated before the value topics. If this is not the case, it would be valuable to have a JSON structure for each value indicating the captureTimestamp for that value, so that I am always sure that they are in sync. However, that would mean a breaking change to the topic structure.

Any advise on the approach to take and how I can ensure my timestamps are correct?

**Comment by tillsteinbach** [comment]
The problem with MQTT is: you never know in which order the messages are delivered. So you need to consider both messages independent.
Would it help you if I add a topic that is triggered once the update is done? that way you could wait with adding the data to the database until all data was received.
What would be also kind of trivial for me to add would be to optionally provide the whole data as a big json
Your proposal with the json for each value would be a bit more work, but also probably not impossible.

**Reply by RaimondB** [reply]
Having a full json snapshot on a separate topic would be great for me. This way i can also make it a batch update to influx converting many values to a measurement at once. :-)

**Reply by tillsteinbach** [reply]
Can you check 0.32.0 with `--with-raw-json-topic` option?

**Reply by RaimondB** [reply]
Now running 0.32.0 and mapping it via node-red, works like a charm. Thanks for the quick implementation :-)

As a future improvement, i think it would be best to combine some of the topics to 1 level higher. See the example below:

` "chargingStatus": {
                        "remainingChargingTimeToComplete_min": 0,
                        "chargingState": "readyForCharging",
                        "chargeMode": "manual",
                        "chargePower_kW": 0.0,
                        "chargeRate_kmph": 0.0,
                        "chargeType": "invalid",
                        "chargingSettings": "default",
                        "carCapturedTimestamp": "2022-03-28T06:24:27+00:00"
                    }`

In this case a topic on the level of **chargingStatus** would be useful, since all values in this node are captured at the same time.
And the capture timestamp can differ in practice across each of these sections where the capturedTimestamp is mentioned.
